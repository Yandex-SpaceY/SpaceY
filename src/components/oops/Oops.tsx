import React, { FC, ReactElement } from 'react';
import { Link } from 'react-router-dom';

import { ROUTE_CONSTANTS } from 'constants/routeConstants';

interface IProps {
  clearState: () => void
}

const Oops: FC<IProps> = ({ clearState }): ReactElement => (
  <>
    <Link to={ROUTE_CONSTANTS.GAME} onClick={clearState}>Go to game</Link>
    <Link to={ROUTE_CONSTANTS.SIGNUP} onClick={clearState}>I want to create an account</Link>
    some other links
  </>
);

export default Oops;
