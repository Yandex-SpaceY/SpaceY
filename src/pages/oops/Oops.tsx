import React from 'react';

const Oops: React.FC = (): React.ReactElement => <h1>Seriously? What are you doing...</h1>;

export default Oops;
