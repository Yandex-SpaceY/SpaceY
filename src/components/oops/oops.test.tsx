import * as React from 'react';
import { shallow } from 'enzyme';

import Oops from './Oops';

it('check render page Oops', () => {
  const component = shallow(<Oops clearState={() => false} />);

  expect(component).toMatchSnapshot();
});
