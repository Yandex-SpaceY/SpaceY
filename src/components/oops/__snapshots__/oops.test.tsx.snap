// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render page Oops 1`] = `
<Fragment>
  <Link
    onClick={[Function]}
    to="/"
  >
    Go to game
  </Link>
  <Link
    onClick={[Function]}
    to="/signup"
  >
    I want to create an account
  </Link>
  some other links
</Fragment>
`;
